libsolidity/parsing/Parser.cpp

```diff
@@ -851,9 +851,16 @@ ASTPointer<ModifierDefinition> Parser::parseModifierDefinition()
 			break;
 	}
 
+	ASTPointer<Block> block;
+	nodeFactory.markEndPosition();
+	if (m_scanner->currentToken() != Token::Semicolon)
```

Comment:
Please try to avoid negations, i.e. use `== Token::Semicolon` and swap the two cases.

Reply:
Make sense. I copied this from `parseFunctionDefinition()` ;)